homeassistant/components/shelly/button.py

```diff
@@ -117,35 +118,25 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set buttons for device."""
-
-    @callback
-    def _async_migrate_unique_ids(
-        entity_entry: er.RegistryEntry,
-    ) -> dict[str, Any] | None:
-        """Migrate button unique IDs."""
-        if TYPE_CHECKING:
-            assert coordinator is not None
-        return async_migrate_unique_ids(entity_entry, coordinator)
-
-    coordinator: ShellyRpcCoordinator | ShellyBlockCoordinator | None = None
+    entry_data = get_entry_data(hass)[config_entry.entry_id]
+    coordinator: ShellyRpcCoordinator | ShellyBlockCoordinator | None
     if get_device_entry_gen(config_entry) in RPC_GENERATIONS:
-        coordinator = get_entry_data(hass)[config_entry.entry_id].rpc
+        coordinator = entry_data.rpc
     else:
-        coordinator = get_entry_data(hass)[config_entry.entry_id].block
-
-    if coordinator is not None:
-        await er.async_migrate_entries(
-            hass, config_entry.entry_id, _async_migrate_unique_ids
-        )
+        coordinator = entry_data.block
 
-        entities: list[ShellyButton] = []
+    if TYPE_CHECKING:
+        assert coordinator is not None
 
-        for button in BUTTONS:
-            if not button.supported(coordinator):
-                continue
-            entities.append(ShellyButton(coordinator, button))
+    await er.async_migrate_entries(
```

Comment:
Need to see if this helper has a time complexity problem.  Unrelated to this PR though

Reply:
🤦 .. we were doing a linear search. Forgot to migrate this one

https://github.com/home-assistant/core/pull/109629